semver/increment.ts

```diff
@@ -73,153 +64,110 @@ export function increment(
-          minor: 0,
-          patch: 0,
-          prerelease: [],
-          build: parseBuild(version.build, build),
-        };
-        break;
-      } else {
-        result = {
-          major: version.major,
-          minor: 0,
-          patch: 0,
-          prerelease: [],
-          build: parseBuild(version.build, build),
-        };
-        break;
-      }
-    case "minor":
-      // If this is a pre-minor version, bump up to the same minor version.
-      // Otherwise increment minor.
+      const isPrerelease = (version.prerelease ?? []).length === 0;
+      const major = isPrerelease || version.minor !== 0 || version.patch !== 0
+        ? version.major + 1
+        : version.major;
+      return {
+        major,
+        minor: 0,
+        patch: 0,
+        prerelease: [],
+        build,
+      };
+    }
+    case "minor": {
+      // If this is a pre-minor version, bump up to the same minor version. Otherwise increment minor.
       // 1.2.0-5 bumps to 1.2.0
       // 1.2.1 bumps to 1.3.0
-      if (
-        version.patch !== 0 ||
-        (version.prerelease ?? []).length === 0
-      ) {
-        result = {
-          major: version.major,
-          minor: version.minor + 1,
-          patch: 0,
-          prerelease: [],
-          build: parseBuild(version.build, build),
-        };
-        break;
-      } else {
-        result = {
-          major: version.major,
-          minor: version.minor,
-          patch: 0,
-          prerelease: [],
-          build: parseBuild(version.build, build),
-        };
-        break;
-      }
-    case "patch":
+      const isPrerelease = (version.prerelease ?? []).length === 0;
+      const minor = isPrerelease || version.patch !== 0
+        ? version.minor + 1
+        : version.minor;
+      return {
+        major: version.major,
+        minor,
+        patch: 0,
+        prerelease: [],
+        build,
+      };
+    }
+    case "patch": {
       // If this is not a pre-release version, it will increment the patch.
       // If it is a pre-release it will bump up to the same patch version.
       // 1.2.0-5 patches to 1.2.0
       // 1.2.0 patches to 1.2.1
-      if ((version.prerelease ?? []).length === 0) {
-        result = {
-          major: version.major,
-          minor: version.minor,
-          patch: version.patch + 1,
-          prerelease: [],
-          build: parseBuild(version.build, build),
-        };
-        break;
-      } else {
-        result = {
-          major: version.major,
-          minor: version.minor,
-          patch: version.patch,
-          prerelease: [],
-          build: parseBuild(version.build, build),
-        };
-        break;
-      }
-    // 1.0.0 "pre" would become 1.0.0-0
-    // 1.0.0-0 would become 1.0.0-1
-    // 1.0.0-beta.0 would be come 1.0.0-beta.1
-    // switching the pre identifier resets the number to 0
-    case "pre":
-      result = {
+      const isPrerelease = (version.prerelease ?? []).length === 0;
```

Comment:
```suggestion
      const isPrerelease = version.prerelease?.length === 0;
```

Reply:
That is not the same statement. 
`version.prerelease?.length === 0` returns false if `version.prerelease` is `undefined`
but
`(version.prerelease ?? []).length === 0` returns true if `version.prerelease`  is `undefined`

Reply:
Yep. Sorry. Will revert.

Reply:
I guess one could write it as `!version.prerelease || version.prerelease.length === 0`